assets/src/scripts/components/graph/state/layout.js

```diff
@@ -0,0 +1,36 @@
+const MOUNT_POINT = 'components/graph/layout';
+const GRAPH_LAYOUT_SET = `${MOUNT_POINT}/GRAPH_LAYOUT_SET`;
+const DEFAULT_LAYOUT = {
+    width: 0,
+    height: 0
+};
+
+export const setLayout = function (layout) {
+    return {
+        type: GRAPH_LAYOUT_SET,
+        payload: {
+            layout
+        }
+    };
+};
+
+export const getLayout = state => state[MOUNT_POINT].layout || DEFAULT_LAYOUT;
+
+export const reducer = function (state = {}, action) {
+    switch (action.type) {
+        case GRAPH_LAYOUT_SET:
+            return {
+                ...state,
+                layout: {
+                    ...state.options,
+                    ...action.payload.layout
+                }
+            };
+        default:
+            return state;
+    }
+};
+
+export default {
+    [MOUNT_POINT]: reducer
+};
```

Comment:
The "duck" concept (https://medium.freecodecamp.org/scaling-your-redux-app-with-ducks-6115955638be) specifies that a reducer be the default export - but I'm finding it more convenient to define the mount point here rather than where `combineReducers` is called, because the selectors need to know where their corresponding state is mounted.

This leads to state that is shaped like this:

```
{
  'components/graph/layout': {
    layout: {
      width: 1089,
      height: 175
    }
  },
  'components/graph/options': {
    component: 'graph',
    siteid: '430406089232901',
    agencycode: 'USGS'
  },
  'services/cache': {
    waterLevels: {
        // ...
      }
    }
  }
}
```

Reply:
I'm wondering if this makes it more difficult to share state between components. For instance, I suspect siteid and agencycode will be needed in other components. In WDFN we needed siteid in both the map and graph components as well as the gageHeight data.

Reply:
My hope is that these ducks will only in rare cases need to import selectors from other components. In the case of `siteid` / `agencycode`, it's used to generate the ID into `waterLevels` (and potentially others), so maybe that's selector worthy - ie, `getCurrentLocationID` could be added to `components/graph/options`.